src/firefly/js/ui/DialogRootContainer.jsx

```diff
@@ -93,22 +133,59 @@ function reRender(dialogs,tmpPopups) {
  */
 function defineDialog(dialogId, dialog) {
     if (!divElement) init();
-    dialogs[dialogId]= <PopupStoreConnection popupPanel={dialog} dialogId={dialogId}/>;
+    const idx= dialogs.findIndex((d) => d.dialogId===dialogId);
+    const newD= {
+        dialogId,
+        component: <PopupStoreConnection popupPanel={dialog} dialogId={dialogId} zIndex={1}/>
+    };
+    if (idx < 0) {
+        dialogs.push(newD);
+    }
+    else {
+        dialogs[idx]= newD;
+    }
+    dialogs= sortZIndex(dialogs,dialogId);
     reRender(dialogs,tmpPopups);
 }
 
+
+/**
+ * Set the zindex for the active one is on top while maintaining the order of the others.
+ * @param inDialogAry
+ * @param topId
+ * @return {*}
+ *
+ *
+ * This is probably not the best way to do this.  I take the active id and make make it so it is bigger
+ * than any of the others. Over time my counter will get too high so occasionally I reduce all the zIndexes too
+ * something more reasonable.
+ */
+function sortZIndex(inDialogAry, topId) {
+    var max= inDialogAry.reduce((prev,d) => (d.zIndex &&d.zIndex>prev) ? d.zIndex : prev, 1);
+    if (max<inDialogAry.length) max= inDialogAry.length+1;
+    var retval= inDialogAry.map( (d,idx) => Object.assign({},d,{zIndex:d.dialogId===topId ? max+1 : (d.zIndex||idx+1) }));
+
+    var min= retval.reduce((prev,d) => (d.zIndex &&d.zIndex<prev) ? d.zIndex : prev, Number.MAX_SAFE_INTEGER);
+    if (retval.length>1 && min>10000) {
+        inDialogAry.forEach( (d) => d.zIndex-=7000);
+    }
+    return retval;
+}
+
+
+
 /**
  * @param popup {object}
  */
 function showTmpPopup(popup) {
     if (!divElement) init();
```

Comment:
it could be but it really just need to be a simple function. I wrote it like that before we had standardized on how to do it.  I will fix it.